### PR TITLE
NMS-13484: Document the Grafana Image Renderer plugin's dependencies

### PR DESCRIPTION
**Pull request by Bonrob2**
NMS-13484: Document the Grafana Image Renderer plugin's dependencies - documented that dependency exists and then link to the full dependency list.

# Pull Request Check Sheet

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new feature, is there documentation?
* [ ] If this is a new feature or substantial change, are there tests that cover it?

# Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

Once the reviewer(s) accept the PR and the branch passes continuous integration in Bamboo, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

# External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/${JIRA-ISSUE-NUMBER}
* Continuous Integration: [CircleCI](https://circleci.com/gh/OpenNMS/opennms-helm)

### CODE REVIEW

**Review by mmahacek — COMMENTED** [review]
I don't see a reference to Puppeter on the image renderer site. https://grafana.com/grafana/plugins/grafana-image-renderer/#installation

**Comment by mmahacek** [comment]
Unless this is a new feature, I'd also suggest foundation-2022 instead of develop.

**Comment by Bonrob2** [comment]
> Unless this is a new feature, I'd also suggest foundation-2022 instead of develop.

It's supposed to be going into Helm develop. Let me check what when wrong.

**Comment by dino2gnt** [comment]
Puppeteer is only related virtue of also using headless chromium and having the good sense to also document headless chromium's complete package dependency list.

**Comment by mmahacek** [comment]
Given this is a NMS issue, I didn't realize it was against the Helm repository.

**Comment by Bonrob2** [comment]
> Given this is a NMS issue, I didn't realize it was against the Helm repository.

No worries. At some point someone suggested adding it to the Helm installation guide instead of the NMS docs.

File: docs/modules/installation/pages/requirements.adoc [hunk 1]
```diff
@@ -6,6 +6,10 @@
 {page-component-title} works with Grafana Server version *7.0* and up.
 It has been tested with Grafana versions up to 7.2 at the time of release.
 
+Note that the Grafana image renderer plugin depends on a project called https://github.com/puppeteer/puppeteer[Puppeteer] for an embedded headless Chrome/Chromium that has some unrequired dependencies.
```
Comment (mmahacek) [inline docs/modules/installation/pages/requirements.adoc:9]: Are we saying that users should install Puppeter, or just the list of dependencies listed in the article on line 10?

**Comment by Bonrob2** [comment]
> Puppeteer is only related by virtue of also using headless chromium and having the good sense to also document headless chromium's complete package dependency list.

So is Puppeteer required, or are we just referencing the complete package dependency list that they have?

File: docs/modules/installation/pages/requirements.adoc (hunk 1 above)
Comment (mmahacek) [reply docs/modules/installation/pages/requirements.adoc:9]: The Grafana renderer doesn't specifically require Puppeteer, but Puppeteer is a headless option (no GUI) that satisfies the requirements.  However, line 10 is referencing a dependency list of packages that Puppeteer requires but doesn't auto-install.  Users need to install those packages before installing Puppeteer.
This isn't immediately clear in the current wording here, and could be confusing to users.  (side note, I setup a docker instance with grafana and the renderer and that was really easy to deploy, but we can't assume all users are running within docker)

Comment (mmahacek) [reply docs/modules/installation/pages/requirements.adoc:9]: Can we just give the same list of packages they mention without installing puppeteer?

Comment (dino2gnt) [reply docs/modules/installation/pages/requirements.adoc:9]: I take that back. According to the original Jira from last year, puppeteer was at some point a dependency of the grafana renderer.  If it isn't a dependency today, then maybe all of this is nonsense. :man_shrugging:

File: docs/modules/installation/pages/requirements.adoc [hunk 2]
```diff
@@ -6,6 +6,10 @@
...
+See Puppeteer's https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md#chrome-headless-doesnt-launch-on-unix[Troubleshooting documentation] for the full dependency list.
+
```
Comment (mmahacek) [inline docs/modules/installation/pages/requirements.adoc:11]: Just replace with a link to Grafana's documentation for the plugin.
```suggestion
The Grafana Dashboard Report feature in Meridian/Horizon requires the
https://grafana.com/grafana/plugins/grafana-image-renderer/[Grafana Image Renderer plugin] to be installed.
```

**Review by mmahacek — APPROVED** [review]
I meant for it to be on one line, but it will render as a single line, so yeah. LGTM.

**Pull request merged**